lightkurve/prf/scenemodels.py

```diff
@@ -0,0 +1,670 @@
+                                  shift_row=self.motion_prior.shift_row.mean[0],
+                                  fitted=self.fit_motion)
+        initial_params = SceneModelParameters(stars=initial_star_guesses,
+                                              background=background,
+                                              focus=focus,
+                                              motion=motion)
+        return initial_params
+
+    def predict(self, params=None):
+        """Returns a synthetic Kepler image given a set of scene parameters.
+
+        Attributes
+        ----------
+        params : ```SceneModelParameters``` object
+            Parameters which define the scene.
+
+        Returns
+        -------
+        synthetic_image : 2D ndarray
+            Predicted image given the parameters.
+        """
+        if params is None:
+            params = self.get_initial_guesses()
+        star_images = []
+        for star in params.stars:
+            star_images.append(self.prfmodel(flux=star.flux,
+                                             center_col=star.col + params.motion.shift_col,
+                                             center_row=star.row + params.motion.shift_row,
+                                             scale_col=params.focus.scale_col,
+                                             scale_row=params.focus.scale_row,
+                                             rotation_angle=params.focus.rotation_angle))
+        synthetic_image = np.sum(star_images, axis=0) + params.background.flux
+        return synthetic_image
+
+    def _predict(self, *params_array):
+        """Wrapper around ``predict()`` which takes an array of shape (n,)
+        where n is the number of free parameters.
+
+        Unlike ``predict()`, this function can be called by scipy.optimize.
+        """
+        params = self.params.from_array(params_array)
+        return self.predict(params)
+
+    def __call__(self, *params_array):
+        return self._predict(*params_array)
+
+    def gradient(self, *params_array):
+        """UNFINISHED WORK!
+        Ze: how to implement this gradient??
+        """
+        params = self.params.from_array(params_array)
+        grad = []
+        for star in params.stars:
+            # Caution: prfmodel takes (flux, col, row) in a different order
+            # than scene model, this has yet to be streamlined
+            grad.append(self.prfmodel.gradient(flux=star.flux,
+                                               center_col=star.col,
+                                               center_row=star.row))
+        # We assume the background gradient is proportional to one
+        grad.append([np.ones(self.prfmodel.shape)])
+        # We assume the gradient of other parameters is zero
+        for i in range(len([*params_array]) - 3 * len(params.stars) - 1):
+            grad.append([np.zeros(self.prfmodel.shape)])
```

Comment:
We don't want a line/column of zeros in our jacobian matrix xD. Other than that, it looks good, I'm working on adding the gradients for the rest of the parameters.

Reply:
I have to put some values because the size of the gradient needs to match the size of the free parameters.  Shall I put it to one for now?

Reply:
Since for now we have gradients available only for ``flux``, ``col``, ``row``, ``shift_row``, ``shift_col``, and ``bkg``, I would probably assume that the other parameters are fixed